docs/architecture-overview.rst

```diff
@@ -0,0 +1,476 @@
+===================================================
+Enterprise Access Service - Architecture Overview
+===================================================
+
+.. note::
+   This document provides a comprehensive overview of the Enterprise Access service architecture 
+   for development teams new to the Open edX ecosystem.
+
+Introduction
+============
+
+Enterprise Access is a Django-based microservice within the Open edX ecosystem that manages enterprise 
+learner access to educational content through various subsidy mechanisms. The service handles:
+
+- **Access Policies**: Rules governing how learners can access content
+- **Content Assignments**: Direct assignment of content to specific learners
+- **Subsidy Requests**: Learner-initiated requests for subsidized access
+- **Customer Billing**: Self-service purchasing workflows via Stripe
+- **Provisioning**: Enterprise customer and subscription setup workflows
+
+The service integrates with multiple external Open edX services and follows event-driven architecture 
+patterns using Kafka for cross-service communication.
+
+System Context
+==============
+
+Enterprise Access sits at the center of the learner credit ecosystem, coordinating between:
+
+.. code-block::
+
+   ┌─────────────────┐    ┌─────────────────┐    ┌─────────────────┐
+   │   Learner       │    │   Admin         │    │ Enterprise      │
+   │   Portal        │    │   Portal        │    │ API Customers   │
+   │   (Frontend)    │    │   (Frontend)    │    │(via API gateway)│
+   └─────────┬───────┘    └─────────┬───────┘    └─────────┬───────┘
+             │                      │                      │
+             └──────────────────────┼──────────────────────┘
```

Comment:
[nit] if you add support tools via the provisioning work it could be a true "Octopus"